test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -388,5 +388,11 @@ spec = Test.group "Dataflow Warnings" <|
         Problems.assume_no_problems <| condition_1 False x3
         Problems.assume_no_problems <| condition_2 False x3
 
+    Test.specify "should only report unique warnings" <|
+        x = 1
+        y = Warning.attach "Foo!" x
+        z = Warning.attach "Bar!" y
+        result = y + z
+        Warning.get_all result . length . should_equal 2
```

Comment:
We can also have a test where you add "Foo" and "Foo" warnings, and the number of warnings in the result is still 1